pandas/core/nanops.py

```diff
@@ -466,14 +476,20 @@ def nanmean(values, axis=None, skipna=True, mask=None):
     the_sum = _ensure_numeric(values.sum(axis, dtype=dtype_sum))
 
     if axis is not None and getattr(the_sum, 'ndim', False):
-        the_mean = the_sum / count
+        with np.errstate(all="ignore"):
+            # suppress division by zero warnings
+            the_mean = the_sum / count
         ct_mask = count == 0
         if ct_mask.any():
             the_mean[ct_mask] = np.nan
     else:
         the_mean = the_sum / count if count > 0 else np.nan
 
-    return _wrap_results(the_mean, dtype)
+    fill_value = None
```

Comment:
this is not appropriate to do here, rather in _wrap_results

Reply:
OK I guess.  That changes the behavior of _wrap_results for the 8 _other_ places that call it though

Reply:
see my above comments. this is completely changing the code flow.